text/0089-tenant-tokens.md

```diff
@@ -0,0 +1,299 @@
+
+##### 2.1.2.4 `apiKeyPrefix` field
+
+`apiKeyPrefix` permits to verify that the signing API key of the Token is known and valid within MeiliSearch. It must contain the first 8 characters of the MeiliSearch API key that generates and signs the Tenant Token.
+
+##### 2.1.2.5 `exp` field
+
+`exp` is used to specify the expiration date of the Tenant Token if needed. The format is a JSON numeric value representing the number of seconds from 1970-01-01T00:00:00Z UTC until the specified UTC date/time, ignoring leap seconds.
+
+##### 2.1.2.6 `searchRules` JSON object
+
+`searchRules` contains the rules to be enforced at search time for all or specific accessible indexes for the signing API Key.
+
+Here are some valid examples in an attempt to cover all possible use cases.
+
+---
+
+> In this case, all indexes on which the signing API Key has permissions are searchable by the tenant token without any restrictions.
+
+```json
+{
+    "searchRules": {
+        "*": {}
+    }
+}
+```
+
+is equivalent to
+
+```json
+{
+    "searchRules": {
+        "*": null
+    }
+}
+```
+
+is equivalent to
+
+```json
+{
+    "searchRules": ["*"] //This notation does not allow the addition of specific rules. The search will just be accessible on all accessibles indexes from the signing API Key for the Tenant Token without specific rules.
+}
+```
+
+---
+
+> In this case, all searchable indexes from the signing API Key are searchable by the tenant token and MeiliSearch will apply the filter rule before applying the search parameters added by the end user.
+
+```json
+{
+    "searchRules": {
+        "*": {
+            "filter": "user_id = 1"
+        }
+    }
+}
+```
+
+---
+
+> In this case, if the `medical_records` index is searchable from the signing API Key, the tenant token can only search in the `medical_records` index. No further rules impact search results on `medical_records`.
+
+```json
+{
+    "searchRules": {
+        "medical_records": {}
+    }
+}
+```
+is equivalent to
+
+```json
+{
+    "searchRules": {
+        "medical_records": null
+    }
+}
+```
+
+is equivalent to
+
+```json
+{
+    "searchRules": ["medical_records"]
+}
+```
+
+---
+
+> In this case, if the `medical_records` index is searchable from the signing API Key, the tenant token can only search in the `medical_records` index, MeiliSearch will apply the filter rule before applying the search parameters added by the end user.
+
+```json
+{
+    "searchRules": {
+        "medical_records": {
+            "filter": "user_id = 1"
+        }
+    }
+}
+```
+
+---
+
+> In this case, if the `medical_records` and `medical_appointments` indexes are searchable from the signing API Key, the tenant token can only search in those indexes. MeiliSearch will apply the filter rule before applying the search parameters added by the end user.
+
+```json
+{
+    "searchRules": {
```

Comment:
Also, it could be good idea to validate a token, by performing a placeholder search with it.

Reply:
Some users have asked for the possibility to hide the `tenant` field that can be used in a `searchRules, which will probably be implemented in the future. With this type of error, an end-user could have the information of the field that allowed the multi-tenancy when it could have been explicitly hidden via a `hideAttributes` attribute.

The tenant tokens will be generated programmatically in the vast majority of cases. It might be useful to include the TenantToken source in the error message to make it clear that the error comes from `searchRules` format and thus facilitate error resolution, but I wonder how this would work for an end-user who is reporting the bug on a field that needs to be `hidden`.

Reply:
Hey @gmourier I understood your suggestion, and I think it is not the best since we will be creating a connection between two different API's and like you said, blocking the future possibilities of this, but I understood that maybe not possible to add the source to the error message either.

I just wanted to raise the concern with you. What do you think to add some sentence in the `Future possibilities` regarding this topic? 
Talking to @curquiza she told me that is best to add a new [discussion in the product](https://github.com/meilisearch/product/discussions/398) in order to keep track of this.

Reply:
Thanks Bruno, I will add it to the Future Possibilities!